froide/foirequest/forms/postal.py

```diff
@@ -142,3 +151,73 @@ def create_attachment(self, message, upload):
             partial(move_upload_to_attachment.delay, att.id, upload.id)
         )
         return att
+
+
+class ThinModelChoiceIterator(ModelChoiceIterator):
+    def __init__(self, field):
+        self.field = field
+        # basically empty the queryset so the thousands of choices are not rendered into context
+        self.queryset = field.queryset[:0]
+
+
+class ThinModelChoiceField(ModelChoiceField):
```

Comment:
I don't quite understand what this does

Reply:
When I serialized the Form with it's `publicbody` field in the template to pass it via prop to vue, it serialized *all* publicbodies into 4MB of JSON (due to `queryset=PublicBody.objects.all()`). Somewhere I found this technique to override the `iterator` to the field so it still works for `save()` and such but doesn't serialize all objects... this is probably not idiomatic or even clean, but the best I could come up with :|

Reply:
@krmax44 I've just looked into this again
- `raw_id_fields` and ForeignKeyRawIdWidget only work for Admin, as far as I can tell
- is using an integer, like [here](https://github.com/okfde/froide/blob/main/froide/document/documents.py#L29) a similar idea? seems hacky
- I suspect this will be moot anyway if/when we turn this into an API?

Reply:
how about a [hidden input widget](https://docs.djangoproject.com/en/5.1/ref/forms/widgets/#hiddeninput)? the linked example (point 2) is about a elasticsearch model, which looks the same but is quite different 🙃